### PR TITLE
feat: [STO-5873]: add coverity security step ui in NG

### PR DESCRIPTION
**Pull request by atmollo-harness**
### Summary

Add Coverity security step in NG

#### Screenshots

Screen Shot 2023-06-27 at 1 49 27 PM

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by jsentongo — DISMISSED** [review]

File: src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStepFunctionConfigs.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  commonFieldsTransformConfig as transformValuesFieldsConfigValues
+} from '../constants'
+import type { Field, InputSetViewValidateFieldsConfig } from '../types'
+import type { CoverityStepData } from './CoverityStep'
+const nonIngestionOnlyFields = ['spec.target.workspace']
+
+export const transformValuesFieldsConfig = (data: CoverityStepData): Field[] =>
+  transformValuesFieldsConfigValues(data).filter(field => !nonIngestionOnlyFields.includes(field.name))
+
+export const editViewValidateFieldsConfig = (data: CoverityStepData) => {
```
Comment (jsentongo) [inline src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStepFunctionConfigs.ts:22]:
```suggestion
export const editViewValidateFieldsConfig = (data: CoverityStepData) : InputSetViewValidateFieldsConfig[] => {
```

File: src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStep.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  initialValues: CoverityStepData
+  template?: CoverityStepData
+  path?: string
+  isNewStep?: boolean
+  readonly?: boolean
+  stepViewType: StepViewType
+  onUpdate?: (data: CoverityStepData) => void
+  onChange?: (data: CoverityStepData) => void
+  allowableTypes: AllowedTypes
+  formik?: any
```
Comment (jsentongo) [inline src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStep.tsx:35]:
```suggestion
  formik?: FormikProps<CoverityStepData>
```

Comment (atmollo-harness) [reply src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStep.tsx]: fixed

File: src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStepFunctionConfigs.ts (hunk 1 above)
Comment (atmollo-harness) [reply src/modules/85-sto/components/PipelineSteps/CoverityStep/CoverityStepFunctionConfigs.ts]: fixed

**Pull request closed without merging**